log4j-web/src/main/java/org/apache/logging/log4j/web/WebLookup.java

```diff
@@ -91,6 +93,18 @@ public String lookup(final LogEvent event, final String key) {
                     HttpServletRequest.class.cast(req).getParameter(name) : null;
         }
 
+        if (key.startsWith(SESSION_ATTR_PREFIX)) {
+            final ServletRequest req = getRequest();
+            if (HttpServletRequest.class.isInstance(req)) {
```

Comment:
Why the reflection? You could change this to be:

```
if (req instanceof HttpServletRequest) {
  final HttpSession session = ((HttpServletRequest) req).getSession(false);
  // ...
}
```

Reply:
Guess it is mainly a coding habit and the preference over a debuggable method (java.lang.Class#cast). Not strong objection if you feel it must be change